services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationDataMapper.kt

```diff
@@ -15,38 +18,93 @@ object ApplicationDataMapper {
                 pendingOnClient = pendingOnClient,
                 identificationNumber = hankeTunnus,
+                customerReference = null,
+                area = null,
+                trafficArrangementImages = null,
                 clientApplicationKind = description, // intentional
                 workDescription = description,
+                propertyDeveloperWithContacts =
+                    propertyDeveloperWithContacts?.toAlluData(
+                        path("propertyDeveloperWithContacts")
+                    ),
                 contractorWithContacts =
                     contractorWithContacts
-                        .orThrow(path("contractorWithContacts"))
+                        .orThrow("contractorWithContacts")
                         .toAlluData(path("contractorWithContacts")),
-                postalAddress = postalAddress?.toAlluData(),
+                constructionWork = constructionWork,
+                maintenanceWork = maintenanceWork,
+                emergencyWork = emergencyWork,
+                propertyConnectivity = propertyConnectivity
+            )
+        }
+
+    fun toAlluData(
+        hankeTunnus: String,
+        applicationData: ExcavationNotificationApplicationData
+    ): AlluExcavationNotificationApplicationData =
+        with(applicationData) {
+            val description =
+                workDescription(
+                    workDescription,
+                    rockExcavation ?: false
+                ) // TODO as in cable report?
+            AlluExcavationNotificationApplicationData(
+                postalAddress =
+                    PostalAddress(
+                        // TODO: this should be a combination of all area addresses (HAI-1542)
+                        streetAddress = StreetAddress(""),
+                        postalCode = "",
+                        city = ""
+                    ),
+                name = name,
+                customerWithContacts =
+                    customerWithContacts
+                        .orThrow("customerWithContacts")
+                        .toAlluData(path("customerWithContacts")),
                 representativeWithContacts =
                     representativeWithContacts?.toAlluData(path("representativeWithContacts")),
-                invoicingCustomer = invoicingCustomer?.toAlluData(path("invoicingCustomer")),
+                invoicingCustomer = invoicingCustomer?.toAlluData("invoicingCustomer"),
+                geometry = getGeometry(this),
+                startTime = startTime.orThrow(path("startTime")),
+                endTime = endTime.orThrow(path("endTime")),
+                pendingOnClient = pendingOnClient,
+                identificationNumber = hankeTunnus,
                 customerReference = customerReference,
-                area = area,
+                area = null,
+                clientApplicationKind = description, // TODO: intentional as in cable report?
```

Comment:
Maybe ask someone before adding it to a PR?

Reply:
Yes, you're right. I asked about this from Allu expert.